arsceneview/src/main/java/io/github/sceneview/ar/node/ArNode.kt

```diff
@@ -145,6 +167,57 @@ open class ArNode() : ModelNode(), ArSceneLifecycleObserver {
         anchor = null
     }
 
+    /**
+     * ### Hosts a Cloud Anchor based on the [anchor]
+     *
+     * The [anchor] is replaced with a new anchor returned by [Session.hostCloudAnchorWithTtl].
+     *
+     * @param ttlDays The lifetime of the anchor in days. See [Session.hostCloudAnchorWithTtl] for more details.
+     * @param onRequestComplete Called when the request completes successfully or with an error.
+     */
+    fun hostCloudAnchor(ttlDays: Int = 1, onRequestComplete: (anchor: Anchor, success: Boolean) -> Unit) {
+        checkRequestState()
+
+        if (anchor == null) throw IllegalStateException("The anchor shouldn't be null")
+
+        anchor = arSession?.hostCloudAnchorWithTtl(anchor, ttlDays)
+        requestInProgress = true
+        this.onRequestComplete = onRequestComplete
+    }
+
+    /**
+     * ### Resolves a Cloud Anchor
+     *
+     * The [anchor] is replaced with a new anchor returned by [Session.resolveCloudAnchor].
+     *
+     * @param cloudAnchorId The Cloud Anchor ID of the Cloud Anchor.
+     * @param onRequestComplete Called when the request completes successfully or with an error.
+     */
+    fun resolveCloudAnchor(cloudAnchorId: String, onRequestComplete: (anchor: Anchor, success: Boolean) -> Unit) {
+        checkRequestState()
+
+        anchor = arSession?.resolveCloudAnchor(cloudAnchorId)
+        requestInProgress = true
+        this.onRequestComplete = onRequestComplete
+    }
+
+    private fun checkRequestState() {
+        if (requestInProgress) throw IllegalStateException("The request is already in progress")
+    }
+
+    /**
+     * ### Cancels a resolve request
+     *
+     * The [anchor] is detached to cancel the resolve request.
+     */
+    fun cancelResolveRequest() {
```

Comment:
```suggestion
cancelCloudAnchorTask() {
```

Reply:
You must also add the call to it inside the `onDestroy()`

Reply:
According to the documentation it's only possible to cancel a resolve request this way: https://developers.google.com/ar/develop/java/cloud-anchors/developer-guide#cancel_or_remove_a_cloud_anchor